test/box-luatest/gh_8051_set_box_cfg_thru_env_test.lua

```diff
@@ -0,0 +1,23 @@
+local server = require('luatest.server')
+
+local t = require('luatest')
+local g = t.group()
+
+local function server_test_nested_table()
+    t.assert_equals(box.cfg.metrics.labels.alias, 'gh_8051')
+
+    t.assert_equals(box.cfg.log_modules, {['aaa'] = 'info', ['bbb'] = 'error'})
+end
+
+g.test_nested_table = function()
+    local env = {["TT_METRICS"] = '{"labels":{"alias":"gh_8051"},' ..
```

Comment:
Json options that starts with '[' must also be tested. I mean if I remove `or raw_value:startswith('[')` from the code above, all tests will still pass. All functionality must be tested.

Reply:
I am not able to create a sane check on json encoding that starts with `'['`, because for now we don't have many `box.cfg{}` options that take in keyless tables (arrays). I can't use `TT_LISTEN` as `luatest.server` ignores it. It seems like this feature uses all the same lines as one with ``{`` and we don't need to test json, so maybe ignore such test? Otherwise, please help me find an option that can obtain keeless tables as value.

Reply:
At least replication takes arrays. But it won't be simple to start a cluster. In any case `popen` will help with TT_LISTEN. We should cover with tests all the code we write.

Reply:
I guess it can be tested with `luatest.server` using an own instance file (`command = <...>`).

The default instance file is just convenient feature introduced less than two years ago (61bf298542420e6cd1496ffe35f1ec5dd84f5e04). Before that the developers have to write own instance file per test (but sometimes they were shared between tests). Of course, nobody is obligated to use the default instance file, which sets `listen` on its own (for convenience in the most cases).

Reply:
Implemented the test via `popen` helper.